tools/apiview/parsers/cpp-api-parser/ParseTests/tests.cpp

```diff
@@ -529,7 +528,42 @@ TEST_F(TestParser, Templates)
 
   auto& db = processor.GetClassesDatabase();
   // Until we get parsing types working correctly, we can't do the syntax check tests.
-//  EXPECT_TRUE(SyntaxCheckClassDb(db, "Template1.cpp"));
+  //  EXPECT_TRUE(SyntaxCheckClassDb(db, "Template1.cpp"));
+}
+
+TEST_F(TestParser, UsingNamespace)
+{
+  ApiViewProcessor processor("tests", R"({
+  "sourceFilesToProcess": [
+    "UsingNamespace.cpp"
+  ],
+  "additionalIncludeDirectories": [],
+  "additionalCompilerSwitches": null,
+  "allowInternal": false,
+  "includeDetail": false,
+  "includePrivate": false,
+  "filterNamespace": null
+}
+)"_json);
+
+  EXPECT_EQ(processor.ProcessApiView(), 0);
+
+  auto& db = processor.GetClassesDatabase();
+  EXPECT_TRUE(SyntaxCheckClassDb(db, "UsingNamespace1.cpp"));
+
+  NsDumper dumper;
+  db->DumpClassDatabase(&dumper);
+  EXPECT_EQ(1ul, dumper.Messages.size());
+
+  size_t usingNamespaces = 0;
+  for (const auto& msg : dumper.Messages)
+  {
+    if (msg.DiagnosticId == "CPA0009")
+    {
+      usingNamespaces += 1;
+    }
+  }
+  EXPECT_EQ(usingNamespaces, 1ul);
 }
 
 #if 0
```

Comment:
Are these two tests every going to be valid?  We should enable or remove.

Reply:
Could you help me understand how this would stop being valid in the future?

Reply:
I was referring to the #if 0 tests further below.  Unfortunately, the PR mechanisms didn't let me highlight those lines.  Its an orthogonal problem unrelated to your PR, but since you're in here I noticed/mentioned it.